nilearn/input_data/fmriprep_confounds.py

```diff
@@ -0,0 +1,300 @@
+        Standardized DVARS threshold for scrub (default = 3).
+        DVARs is defined as root mean squared intensity difference of volume N
+        to volume N+1 :footcite:`Power2012`. D referring to temporal derivative
+        of timecourses, VARS referring to root mean squared variance over
+        voxels.
+
+    wm_csf : {'basic', 'power2', 'derivatives', 'full'}
+        Type of confounds extracted from masks of white matter and
+        cerebrospinal fluids.
+
+        - "basic" the averages in each mask (2 parameters)
+        - "power2" averages and quadratic terms (4 parameters)
+        - "derivatives" averages and derivatives (4 parameters)
+        - "full" averages + derivatives + quadratic terms + power2d derivatives
+          (8 parameters)
+
+    global_signal : {'basic', 'power2', 'derivatives', 'full'}
+        Type of confounds extracted from the global signal.
+
+        - "basic" just the global signal (1 parameter)
+        - "power2" global signal and quadratic term (2 parameters)
+        - "derivatives" global signal and derivative (2 parameters)
+        - "full" global signal + derivatives + quadratic terms + power2d
+          derivatives (4 parameters)
+
+    scrub : {'full', 'basic'}
+        Type of scrub of frames with excessive motion :footcite:`Power2014`
+
+        - "basic" remove time frames based on excessive framewise displacement
+          and DVARS.
+        - "full" also remove continuous segments containing fewer than 5
+          volumes.
+
+        One-hot encoding vectors are added as regressors for each scrubbed
+        frame.
+
+    compcor : {'anat_combined', 'anat_separated', 'temporal',
+    'temporal_anat_combined', 'temporal_anat_separated'}
+
+        .. warning::
+            Require fmriprep >= v:1.4.0.
+
+        Type of confounds extracted from a component based noise correction
+        method :footcite:`BEHZADI200790`.
+
+        - "anat_combined" noise components calculated using a white matter and
+          CSF combined anatomical mask
+        - "anat_separated" noise components calculated using white matter mask
+          and CSF mask compcor separately; two sets of scores are concatenated
+        - "temporal" noise components calculated using temporal compcor
+        - "temporal_anat_combined" components of "temporal" and "anat_combined"
+        - "temporal_anat_separated" components of "temporal" and
+          "anat_separated"
+
+    n_compcor : "all" or int, default "all"
+        The number of noise components to be extracted.
+        For acompcor_combined=False, and/or compcor="full", this is the number
+        of components per mask.
+        "all": select all components (50% variance explained by fMRIPrep
+        defaults)
+
+    ica_aroma : {'full', 'basic'}
+
+        - "full": use fMRIprep output `~desc-smoothAROMAnonaggr_bold.nii.gz`.
+        - "basic": use noise independent components only.
+
+    demean : boolean, default True
```

Comment:
I initially thought that this should be renamed to `standardize_confounds` to match nilearn's naming, but looking at the function `_demean_confounds` it seems that only the mean of kept volumes is subtracted. 
But, I believe that when `signal.clean` has `standardize_confounds` set to True, it also standardizes wrt std.
Do you think this difference could be problematic?

Reply:
> But, I believe that when `signal.clean` has `standardize_confounds` set to True, it also standardizes wrt std.
> Do you think this difference could be problematic?

I have the same question as well, and that definitely seems weird to me.

I tried to demean the confounds and set `standardize_confounds=False`, and everything got weird....

Reply:
I think I have found the reason why some combinations of parameters fails. It's related to how `detrend` works in `signal.clean`. I will start with a TL;DR - if we want `fmriprep_confounds` to work with `signal.clean` default settings correctly, the default value of the `demean` option should be `False`. Otherwise the confounds would be demeaned twice. 

For `fmriprep_confounds`, `demean=False`, all combination of the relevant parameters in `signal.clean`:
|standardize_confounds|standardize|detrend|test_fmriprep_confounds|
|--|--|--|--|
|True|zscore|True|PASSED|
|True|zscore|linear|PASSED|
|True|zscore|False|PASSED|
|True|psc|True|FAILED|
|True|psc|linear|FAILED|
|True|psc|False|PASSED|
|True|False|True|PASSED|
|True|False|linear|PASSED|
|True|False|False|PASSED|
|False|zscore|True|PASSED|
|False|zscore|linear|PASSED|
|False|zscore|False|FAILED|
|False|psc|True|FAILED|
|False|psc|linear|FAILED|
|False|psc|False|FAILED|
|False|False|True|PASSED|
|False|False|linear|PASSED|
|False|False|False|FAILED|

AFAIK the `signal.clean` is doing what it's supposed to do, my intuition is some combination of options probably shouldn't be allowed. This issue is a bit out of the scope of `fmriprep_denoise` but something for `signal.clean`.